### PR TITLE
Add type mapping configuration section

### PR DESCRIPTION
**Pull request by mosabua**
As a follow up we also have to add type mapping info for a bunch of connectors but that should be done separately in another PR. 

Fixes https://github.com/trinodb/trino/issues/7825

As discussed on slack with @losipiuk

### CODE REVIEW

File: docs/src/main/sphinx/connector/oracle.rst [hunk 1]
```diff
@@ -273,7 +266,7 @@ Mapping character types
 ^^^^^^^^^^^^^^^^^^^^^^^
 
 Trino's ``VARCHAR(n)`` maps to ``VARCHAR2(n CHAR)`` if ``n`` is no greater than
-4000. A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
+1.    A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
```
Comment (electrum) [inline docs/src/main/sphinx/connector/oracle.rst:269]: This seems like an unrelated and incorrect change

**Review by electrum — COMMENTED** [review]
One comment, otherwise looks good

**Review by losipiuk — COMMENTED** [review]
LGTM % David's comment

**Comment by losipiuk** [comment]
Mind red CI - related

File: docs/src/main/sphinx/connector/oracle.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/oracle.rst:269]: Wow .. that was so weird. Visual Code thought this is the start of a list and automatically made it the 1. point.. fixed now

**Comment by mosabua** [comment]
Comments addressed and CI fixed.

**Review by losipiuk — APPROVED** [review]

**Comment by mosabua** [comment]
Updated as discussed @electrum .. please merge.

And maybe we should discuss with @losipiuk how the properties are related in the implementation and how we should update the docs after we understand.

**Comment by mosabua** [comment]
Ping @electrum @findepi

**Review by electrum — APPROVED** [review]
